packages/data/src/world/mq/ganon_castle_mq.yml

```diff
@@ -35,92 +39,235 @@
-    "MQ Ganon Castle Shadow Trial Switch Chest": "can_use_bow && has_lens && (has_hover_boots || (can_hookshot && has_fire))"
-    "MQ Ganon Castle SR Shadow Front-Right": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot))"
-    "MQ Ganon Castle SR Shadow Middle": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot))"
-    "MQ Ganon Castle SR Shadow Back-Left": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot && has_fire))"
-    "MQ Ganon Castle SR Shadow Back-Center": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot && has_fire))"
-    "MQ Ganon Castle SR Shadow Front-Center": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot))"
-    "MQ Ganon Pot Shadow End 1": "has_lens && (has_hover_boots || (can_hookshot && has_fire && can_use_bow)) && silver_rupees_ganon_shadow"
-    "MQ Ganon Pot Shadow End 2": "has_lens && (has_hover_boots || (can_hookshot && has_fire && can_use_bow)) && silver_rupees_ganon_shadow"
-    "MQ Ganon Castle Wonder Item": "has_lens && (has_hover_boots || (can_use_bow && can_hookshot)) && (can_use_bow || has_explosives)"
+    "MQ Ganon Castle Spirit Trial Back Right Sun Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+    "MQ Ganon Castle Spirit Trial Back Left Sun Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+    "MQ Ganon Castle Spirit Trial Front Left Sun Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+    "MQ Ganon Castle Spirit Trial Gold Gauntlets Chest": "event(GANON_MQ_SPIRIT_ENDING_SUNS)"
+"Ganon Castle Spirit Trial Ending Room":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Spirit Trial Sun Room Sun Side": "true"
+  events:
+    GANON_TRIAL_SPIRIT: "can_use_light_arrows"
+  locations:
+    "MQ Ganon Pot Spirit End 1": "true"
+    "MQ Ganon Pot Spirit End 2": "true"
+"Ganon Castle Shadow Trial Entrance Side":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Shadow Trial Middle Beamos Area 1": "event(GANON_MQ_SHADOW_PROGRESSION)"
+    "Ganon Castle Main Room": "true"
+  events:
+    GANON_MQ_SHADOW_BOMBFLOWER: "can_use_bow || (event(GANON_MQ_SHADOW_PROGRESSION) && (can_use_din || has_bombflowers))"
+    GANON_MQ_SHADOW_PROGRESSION: "longshot_anywhere || (has_lens && ((can_hookshot && event(GANON_MQ_SHADOW_BOMBFLOWER)) || has_hover_boots))"
+  locations:
+    "MQ Ganon Castle Shadow Trial Bomb Flower Chest": "event(GANON_MQ_SHADOW_BOMBFLOWER) && (can_hookshot || has_hover_boots)"
+    "MQ Ganon Castle SR Shadow Front-Right": "event(GANON_MQ_SHADOW_PROGRESSION)"
+    "MQ Ganon Castle SR Shadow Front-Center": "event(GANON_MQ_SHADOW_PROGRESSION)"
+    "MQ Ganon Castle Wonder Item": "event(GANON_MQ_SHADOW_BOMBFLOWER) && event(GANON_MQ_SHADOW_PROGRESSION)"
+"Ganon Castle Shadow Trial Middle Beamos Area 1":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Shadow Trial Entrance Side": "event(GANON_MQ_SHADOW_PROGRESSION)"
```

Comment:
From this side, the bombflower event would not be needed, nor would hookshot actually get you to the entrance without anywhere. Also might be worth adding the bombflower event and wonder item on this side too since you then would not need hook or hover to get to the bombflower.

Reply:
Changed with new event.